src/index.ts

```diff
@@ -248,21 +251,23 @@ class Compiler {
   }
 
   private validateEnumMembers(rule: JSONSchema.Schema): EnumType {
-    if (!rule.type) rule.type = 'string'
+    if (rule.type === undefined) rule.type = 'string'
```

Comment:
I don't think this is correct. It is valid to _not_ set a type here. If we default to `'string'` and our enum is `[ false ]`, this would be incorrect.